src/charm.py

```diff
@@ -401,6 +408,62 @@ def _is_magmad_enabled(self) -> bool:
         )
         return not magma_service.returncode
 
+    def _install_configurations(self, event: OrchestratorAvailableEvent) -> bool:
+        """Install or update configuration files.
+
+        Returns:
+            True if any changes were applied
+        """
+        config = self._generate_config(event)
+        return any(
+            [
+                self._install_file(Path(ROOT_CA_PATH), event.root_ca_certificate),
+                self._install_file(Path(CONFIG_PATH), config),
+            ]
+        )
+
+    @staticmethod
+    def _install_file(file: Path, content: str) -> bool:
+        """Install file with provided text content.
+
+        Args:
+            file: Path object to write to
+            content: Text content to write to the file
+
+        Returns:
+            True if the file was written to
+        """
+        if not file.parent.exists():
+            file.parent.mkdir()
+        elif file.exists() and file.read_text() == content:
+            return False
+        file.write_text(content)
+        return True
+
+    @staticmethod
+    def _generate_config(event: OrchestratorAvailableEvent) -> str:
+        return (
+            f"cloud_address: {event.orchestrator_address}\n"
+            f"cloud_port: {event.orchestrator_port}\n"
+            f"bootstrap_address: {event.bootstrapper_address}\n"
+            f"bootstrap_port: {event.bootstrapper_port}\n"
+            f"fluentd_address: {event.fluentd_address}\n"
+            f"fluentd_port: {event.fluentd_port}\n"
+            "\n"
+            f"rootca_cert: {ROOT_CA_PATH}\n"
+        )
+
+    @staticmethod
+    def _restart_magma() -> None:
+        subprocess.run(
+            ["service", "magma@*", "stop"],
+            stdout=subprocess.PIPE,
+        )
+        subprocess.run(
+            ["service", "magma@magmad", "start"],
```

Comment:
Why not simply use `restart` instead of `stop` and `start`

Reply:
Ahh ok forget it then, I thought restart was simply stop + start anyway.

Reply:
I think I was checking that while working on the snap and iirc it's important for Magma services to be started via `magmad`. `magmad` however can't stop subordinate services. That's why 2-step restart.